framework/audio/engine/internal/audioplayer.cpp

```diff
@@ -280,3 +280,30 @@ void AudioPlayer::onMidiStatusChanged(engine::IAudioEngine::Status status)
         onStop();
     }
 }
+
+void AudioPlayer::playMidi(const midi::MidiData& data)
+{
+    if (!m_singleNoteMidiSource) {
```

Comment:
I think it'd be better to not mention anything related with "notes" here

Reply:
I agree, in one PR this will be fixed